database-commons/src/main/java/io/cdap/plugin/db/batch/config/AbstractDBSpecificSourceConfig.java

```diff
@@ -0,0 +1,246 @@
+  @Description("Bounding Query should return the min and max of the " +
+    "values of the 'splitBy' field. For example, 'SELECT MIN(id),MAX(id) FROM table'. " +
+    "This is required unless numSplits is set to one.")
+  @Macro
+  private String boundingQuery;
+
+  @Nullable
+  @Name(SPLIT_BY)
+  @Description("Field Name which will be used to generate splits. This is required unless numSplits is set to one.")
+  @Macro
+  private String splitBy;
+
+  @Nullable
+  @Name(NUM_SPLITS)
+  @Description("The number of splits to generate. If set to one, the boundingQuery is not needed, " +
+    "and no $CONDITIONS string needs to be specified in the importQuery. If not specified, the " +
+    "execution framework will pick a value.")
+  @Macro
+  private Integer numSplits;
+
+  @Nullable
+  @Name(SCHEMA)
+  @Description("The schema of records output by the source. This will be used in place of whatever schema comes " +
+    "back from the query. This should only be used if there is a bug in your jdbc driver. For example, if a column " +
+    "is not correctly getting marked as nullable.")
+  private String schema;
+
+  public String getImportQuery() {
+    return cleanQuery(importQuery);
+  }
+
+  public String getBoundingQuery() {
+    return cleanQuery(boundingQuery);
+  }
+
+  public void validate(FailureCollector collector) {
+    boolean hasOneSplit = false;
+    if (!containsMacro(NUM_SPLITS) && numSplits != null) {
+      if (numSplits < 1) {
+        collector.addFailure(
+          String.format("Invalid value for Number of Splits '%d'. Must be at least 1.", numSplits),
+          "Specify a Number of Splits no less than 1.")
+          .withConfigProperty(NUM_SPLITS);
+      }
+      if (numSplits == 1) {
+        hasOneSplit = true;
+      }
+    }
+
+    if (getTransactionIsolationLevel() != null) {
+      TransactionIsolationLevel.validate(getTransactionIsolationLevel(), collector);
+    }
+
+    if (!containsMacro(IMPORT_QUERY) && Strings.isNullOrEmpty(importQuery)) {
+      collector.addFailure("Import Query is empty.", "Specify the Import Query.")
+        .withConfigProperty(IMPORT_QUERY);
+    }
+
+    if (!hasOneSplit && !containsMacro(IMPORT_QUERY) && !getImportQuery().contains("$CONDITIONS")) {
+      collector.addFailure(String.format(
+        "Import Query %s must contain the string '$CONDITIONS'. if Number of Splits is not set to 1.", importQuery),
+                           "Include '$CONDITIONS' in the Import Query")
+        .withConfigProperty(IMPORT_QUERY);
+    }
+
+    if (!hasOneSplit && !containsMacro("splitBy") && (splitBy == null || splitBy.isEmpty())) {
+      collector.addFailure("Split-By Field Name must be specified if Number of Splits is not set to 1.",
+                           "Specify the Split-by Field Name.").withConfigProperty(SPLIT_BY)
+        .withConfigProperty(NUM_SPLITS);
+    }
+
+    if (!hasOneSplit && !containsMacro("boundingQuery") && (boundingQuery == null || boundingQuery.isEmpty())) {
```

Comment:
Should it be `containsMacro(BOUNDING_QUERY)`?

Reply:
good catch, done :)